libsolidity/formal/CHC.cpp

```diff
@@ -739,117 +737,35 @@ set<frontend::Expression const*, CHC::IdCompare> CHC::transactionAssertions(ASTN
 	return assertions;
 }
 
-vector<smtutil::SortPointer> CHC::stateSorts(ContractDefinition const& _contract)
-{
-	return applyMap(
-		SMTEncoder::stateVariablesIncludingInheritedAndPrivate(_contract),
-		[](auto _var) { return smt::smtSortAbstractFunction(*_var->type()); }
-	);
-}
-
 smtutil::SortPointer CHC::constructorSort()
 {
 	solAssert(m_currentContract, "");
-	if (auto const* constructor = m_currentContract->constructor())
-		return sort(*constructor);
-
-	return make_shared<smtutil::FunctionSort>(
-		vector<smtutil::SortPointer>{smtutil::SortProvider::uintSort} + m_stateSorts,
-		smtutil::SortProvider::boolSort
-	);
+	return smt::constructorSort(*m_currentContract);
 }
```

Comment:
Maybe you can delete this one as well, and move `assert` to `smt::constructorSort` along with changing parameter to be pointer?

Reply:
I would prefer to keep the necessary arguments as references in `PredicateSort`. I think it's fine to remove it and just pass `*m_currentContract` in the places that currently call `constructorSort()`, since `m_currentContract` should not be null there.

Reply:
Sounds good